std/file.d

```diff
@@ -3047,14 +3047,23 @@ private struct DirIteratorImpl
         }
     }
 
-    this(string pathname, SpanMode mode, bool followSymlink)
+    this(R)(R pathname, SpanMode mode, bool followSymlink)
+        if (isInputRange!R && isSomeChar!(ElementEncodingType!R))
     {
         _mode = mode;
         _followSymlink = followSymlink;
         _stack = appender(cast(DirHandle[])[]);
-        if(_mode == SpanMode.depth)
+        if (_mode == SpanMode.depth)
             _stashed = appender(cast(DirEntry[])[]);
-        if(stepIn(pathname))
+
+        static if (isNarrowString!R && is(Unqual!(ElementEncodingType!R) == char))
+            alias pathnames = pathname;
+        else
+        {
+            import std.array;
+            string pathnames = pathname.array;
+        }
+        if (stepIn(pathnames))
```

Comment:
pathnames seems to indicate an array of path names which it isn't. Maybe pathnameStr or some such